### DISCUSSION TITLE
Number formatting

### DISCUSSION THREAD

**Discussion by sanyi9305** [uNmAnNeR/imaskjs#620] [Q&A] [answered]
Hi,

I Use a Number mask with scale:2 and overwrite: true.
If I would like write the integer part I cant write more then default value.
Eg: The value is 12.33
And I would like change to 123.33 I Cant make this.
Can you help me how to use?

How can I make it happen that if I press the +/- sign anywhere then my dynamic mask will switch to the right one?

If I use a number mask then if it can be negative and type -1234567
then the input value will be: -1,237,465 is the position of the cursor wrong for a negative number?

If I use thousand separator and signed is true when I press the "-" again and again the cursor position is changing.
And got this error when the cursor in the 0 position and press again:
![image](https://user-images.githubusercontent.com/32463572/154523504-8344c6e2-f080-4514-a031-70725a20ebbd.png)

I attach my maskoptions.

Thanks
[mask.txt](https://github.com/uNmAnNeR/imaskjs/files/8069810/mask.txt)
.

**Comment by uNmAnNeR** [comment, marked as answer]
fixed in v6.4.3
